src/components/shared/ButtonsLarge/index.tsx

```diff
@@ -0,0 +1,34 @@
+import classnames from 'classnames';
+
+export enum ButtonLargeVariants {
+  Primary = 'primary',
+  Outline = 'outline',
+}
+
+interface ButtonLargeProps {
+  variant: string;
+  text: string;
+  hasIcon: boolean;
+  disabled?: boolean;
+  onClick: () => unknown;
+}
+
+const ButtonLarge = ({ variant, text, hasIcon, onClick, disabled }: ButtonLargeProps) => {
+  const variants = classnames(
+    'rounded-lg w-72 h-11 border font-semibold disabled:bg-light-grey disabled:text-dark-grey disabled:outline-none active:text-white focus:outline-focus focus:outline-dashed active:bg-active active:outline-active-outline active:outline active:outline-2',
+    {
+      'bg-dark-violet text-white hover:bg-hover': variant === ButtonLargeVariants.Primary,
+      'bg-white border-dark-violet text-dark-violet hover:bg-hover hover:border-0 hover:text-white disabled:border-grey':
+        variant === ButtonLargeVariants.Outline,
+    },
+  );
+  return (
+    <>
+      <button className={variants} onClick={onClick} disabled={disabled}>
```

Comment:
And here you can pass the spreaded props which will be valid html button props since you are extending the HTML Button type..
```suggestion
      <button className={variants} {...rest}>
        {children}
      </button>
```

Later on you can instantiate your ButtonLarge almost as if it where a normal html button!
```
<ButtonLarge
  variant="yourVariant"
  disabled={yourCondition}
  onClick={yourFunc}
  // any other prop that a html button could have
>
  My Button Content!
</Button>
```